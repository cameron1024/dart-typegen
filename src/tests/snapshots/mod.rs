// #[test]
// fn test_foo() {
//     let dart = include_str!("./foo.dart");
//     let toml = include_str!("./foo.toml");
//
// }
