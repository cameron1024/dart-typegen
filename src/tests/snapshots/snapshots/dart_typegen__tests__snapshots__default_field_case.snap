---
source: src/tests/snapshots/mod.rs
expression: dart
---
final class Foo {
  final String multipleWords;
  final String evenMoreMultipleWords;

  const Foo({required this.multipleWords, required this.evenMoreMultipleWords});

  FooBuilder toBuilder() => FooBuilder(
    multipleWords: multipleWords,
    evenMoreMultipleWords: evenMoreMultipleWords,
  );

  Map<String, dynamic> toJson() => {
    "multipleWords": multipleWords,
    "evenMoreMultipleWords": evenMoreMultipleWords,
  };
  factory Foo.fromJson(Map<String, dynamic> json) => Foo(
    multipleWords: json["multipleWords"] as String,
    evenMoreMultipleWords: json["evenMoreMultipleWords"] as String,
  );

  @override
  String toString() =>
      "Foo("
      "multipleWords: $multipleWords, "
      "evenMoreMultipleWords: $evenMoreMultipleWords"
      ")";
  @override
  bool operator ==(Object other) {
    if (identical(this, other)) {
      return true;
    }
    if (other is! Foo) {
      return false;
    }
    if (multipleWords != other.multipleWords) return false;
    if (evenMoreMultipleWords != other.evenMoreMultipleWords) return false;
    return true;
  }

  @override
  int get hashCode => Object.hashAll([multipleWords, evenMoreMultipleWords]);
}

/// Builder class for [Foo]
final class FooBuilder {
  String multipleWords;
  String evenMoreMultipleWords;

  FooBuilder({
    required this.multipleWords,
    required this.evenMoreMultipleWords,
  });

  Foo build() => Foo(
    multipleWords: multipleWords,
    evenMoreMultipleWords: evenMoreMultipleWords,
  );
}
